Revit_Core_Engine/Convert/Physical/ToRevit/Rebar.cs

```diff
@@ -0,0 +1,125 @@
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using Autodesk.Revit.DB.Structure;
+using BH.Engine.Adapters.Revit;
+using BH.oM.Adapters.Revit.Settings;
+using BH.oM.Physical.Reinforcement;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /****            Interface methods              ****/
+        /***************************************************/
+
+        public static Rebar IToRevitRebar(this IReinforcingBar reinforcement, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            return ToRevitRebar(reinforcement as dynamic, document, settings, refObjects);
+        }
+
+
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        public static Rebar ToRevitRebar(this PrimaryReinforcingBar bar, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            return ToRevitRebar((IReinforcingBar)bar, document, settings, refObjects);
+        }
+
+        /***************************************************/
+
+        public static Rebar ToRevitRebar(this Stirrup stirrup, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            return ToRevitRebar((IReinforcingBar)stirrup, document, settings, refObjects);
+        }
+
+
+        /***************************************************/
+        /****              Private methods              ****/
+        /***************************************************/
+
+        private static Rebar ToRevitRebar(this IReinforcingBar bar, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            if (bar == null || document == null)
+                return null;
+
+            Rebar rebar = refObjects.GetValue<Rebar>(document, bar.BHoM_Guid);
+            if (rebar != null)
+                return rebar;
+
+            settings = settings.DefaultIfNull();
+
+            //getting bar type
+            string barTypeName = ((int)(bar.Diameter * 1000)).ToString();
```

Comment:
I believe it would be good to use `BH.Revit.Engine.Core.Query.ElementType` here (possibly a new public method could be written to call `ElementType(this IBHoMObject bHoMObject, Document document, IEnumerable<BuiltInCategory> builtInCategories, RevitSettings settings = null)`).